onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -1092,10 +1137,66 @@ common::Status DenseTensorToSparseTensorProto(const ONNX_NAMESPACE::TensorProto&
                       IsZero<uint8_t>, CopyElement<uint8_t>, values, indices);
       break;
     }
+    case 2: {
+      // bytes
+      switch (data_type) {
+        case TensorProto_DataType_FLOAT16: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<BFloat16>, CopyElement<uint16_t>, values, indices);
+          break;
+        }
+        case TensorProto_DataType_BFLOAT16: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<MLFloat16>, CopyElement<uint16_t>, values, indices);
+          break;
+        }
+        case TensorProto_DataType_INT16:
+        case TensorProto_DataType_UINT16: {
+          SparsifyGeneric(dense_raw_data.get(), n_dense_elements, element_size,
+                          IsZero<uint16_t>, CopyElement<uint16_t>, values, indices);
+          break;
+        }
+        default:
+          ORT_THROW(false, "BUG! Report to onnxruntime team.");
```

Comment:
>ORT_THROW [](start = 10, length = 9)

More graceful to use ORT_RETURN_IF in a build with exceptions disabled (returns Status vs. abort() being called).